Sources/Base/Types.swift

```diff
@@ -166,20 +192,20 @@ func ==(lhs: DataType, rhs: DataType) -> Bool {
 }
 
 class Decl: ASTNode {
-  var type: DataType
+  var type: DataType?
```

Comment:
might it make more sense to introduce an `error` or `unknown` type instead of making this optional?